notifications/src/main/java/com/dojo/notifications/service/grpc/LeaderboardClient.java

```diff
@@ -0,0 +1,108 @@
+
+    private final ApiMockLeaderboardServiceGrpc.ApiMockLeaderboardServiceBlockingStub leaderboardServiceBlockingStub;
+    private final ApiMockLeaderboardServiceGrpc.ApiMockLeaderboardServiceStub leaderboardServiceStub;
+
+    @Autowired
+    public LeaderboardClient(LeaderboardNotifierService leaderboardNotifierService, ApiMockLeaderboardServiceGrpc.ApiMockLeaderboardServiceBlockingStub leaderboardServiceBlockingStub, ApiMockLeaderboardServiceGrpc.ApiMockLeaderboardServiceStub leaderboardServiceStub) {
+        this.leaderboardNotifierService = leaderboardNotifierService;
+
+        this.leaderboardServiceBlockingStub = leaderboardServiceBlockingStub;
+        this.leaderboardServiceStub = leaderboardServiceStub;
+    }
+
+    public void startLeaderboardNotifications(Contest contest) {
+        String contestId = contest.getContestId();
+        subscribeForNotifications(contestId);
+        getNotifications(contest);
+    }
+
+    public void stopLeaderboardNotifications(String contestId) {
+        final StopRequest request = StopRequest.newBuilder()
+                .setContestId(contestId)
+                .build();
+
+        StopResponse response = leaderboardServiceBlockingStub.stopNotifications(request);
+        LOGGER.info(NOTIFICATION_STOPPED_MESSAGE, contestId, response);
+    }
+
+    private void getNotifications(Contest contest) {
+        final LeaderboardRequest leaderboardRequest = LeaderboardRequest.newBuilder()
+                .setContestId(contest.getContestId())
+                .build();
+
+        leaderboardServiceStub.getLeaderboard(leaderboardRequest, new StreamObserver<LeaderboardResponse>() {
+            @Override
+            public void onNext(LeaderboardResponse leaderboardResponse) {
+                List<Participant> participants = new ArrayList<>();
+
+                leaderboardResponse.getParticipantList().forEach(participantResponse -> {
+                    Participant participant = new Participant();
+                    UserInfo userInfo = new UserInfo();
+                    userInfo.setId(participantResponse.getUser().getId());
+                    userInfo.setName(participantResponse.getUser().getName());
+                    participant.setUser(userInfo);
+                    participant.setScore(participantResponse.getScore());
+
+                    participants.add(participant);
+                });
+
+                leaderboardNotifierService.lookForLeaderboardChanges(contest, new Leaderboard(participants));
+                LOGGER.info(RESPONSE_MESSAGE, leaderboardResponse);
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                LOGGER.error(ERROR_MESSAGE, leaderboardRequest, throwable);
+            }
+
+            @Override
+            public void onCompleted() {
+                LOGGER.info(COMPLETED_MESSAGE);
+            }
+        });
+    }
+
+    private void subscribeForNotifications(String contestId) {
```

Comment:
subscribeForNotifications should be before getNotifications by order as it is called first (e.g. closer to the call)